### ISSUE TITLE
[BUG][Guest cluster] Permission denied for workload with security context

### ISSUE THREAD

**Issue by yasker** [harvester/harvester#1630]
To reproduce:
1. Add Bitname Repo (https://charts.bitnami.com/bitnami) to the guest cluster.
2. Set storage class `harvester` as `default` to workaround bug https://github.com/harvester/harvester/issues/1629.
3. Install `Wordpress` using default options.
4. Manually modify the load balancer for Wordpress to include the healthcheck port `80`.
5. Wait for the Pod to come up.

Result:
The mariadb pod failed to get up, with the error message.

```
mariadb 01:43:54.54
mariadb 01:43:54.54 Welcome to the Bitnami mariadb container
mariadb 01:43:54.54 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb 01:43:54.54 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb 01:43:54.54
mariadb 01:43:54.54 INFO ==> ** Starting MariaDB setup **
mariadb 01:43:54.56 INFO ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb 01:43:54.57 INFO ==> Initializing mariadb database
mkdir: cannot create directory '/bitnami/mariadb/data': Permission denied
```

I think something is wrong with the Harvester guest CSI driver here.

**Comment by yasker** [comment]
Might be the same reason as https://github.com/longhorn/longhorn/issues/2964

**Comment by gitlawr** [comment]
Should be addressed in https://github.com/harvester/harvester/issues/1369

**Comment by guangbochen** [comment]
@yasker what's your guest k8s version with either the RKE1/RKE2 node driver?

**Comment by yasker** [comment]
Thanks @gitlawr @guangbochen , confirmed it's fixed in `v1.21.7+rke2r1`.
